File: my_modules/downloader.js
const download = require('download');
const axios = require('axios');
const config = require('../config');

const BASE_URL = 'https://api.telegram.org';
const BASE_DIR = `${__dirname}/../user_data`;

const downloader = (ctx, fileType) => new Promise((resolve, reject) => {
  const userId = ctx.update.message.from.id;
  const fileId = fileType === 'photo' ? ctx.update.message[fileType][0].file_id : ctx.update.message[fileType].file_id;
  const url = `bot${config.BOT_TOKEN}/getFile?file_id=${fileId}`;

  axios({
    baseURL: BASE_URL,
    url,
    method: 'get',
    responseType: 'json',
  })
    .then((result) => {
      const filePath = result.data.result.file_path;
      const fileName = filePath.split('/')[1];
      const url = `file/bot${config.BOT_TOKEN}/${filePath}`;

      axios({
        baseURL: BASE_URL,
        url,
        method: 'get',
        responseType: 'blob',
      })
        .then(() => {
          download(`${BASE_URL}/${url}`, `${BASE_DIR}/${userId}`)
            .then(() => {
              resolve({
                message: `File successfully downloaded in ${this.downloadPath}`,
                downloadPath: `${BASE_DIR}/${userId}`,
                fileName,
              });
            })
            .catch((err) => {
              reject(new Error(`Error downloading the file: ${err.name}: ${err.message}`));
            });
        })
        .catch((err) => {
          reject(new Error(`Error downloading the file: ${err.name}: ${err.message}`));
        });
    })
    .catch((err) => {
      reject(new Error(`Error downloading the file: ${err.name}: ${err.message}`));
    });
});

module.exports = downloader;

File: downloader.py
from telegram.ext import CallbackContext


def download_file(user_id: int, file_to_download, file_type: str, context: CallbackContext) -> str:
    user_download_dir = f"downloads/{user_id}"
    file_id = ''
    file_extension = ''

    if file_type == 'audio':
        file_id = context.bot.get_file(file_to_download.file_id)
        file_name = file_to_download.file_name
        file_extension = file_name.split(".")[-1]
    elif file_type == 'photo':
        file_id = context.bot.get_file(file_to_download.file_id)
        file_extension = 'jpg'

    file_download_path = f"{user_download_dir}/{file_id.file_id}.{file_extension}"

    try:
        file_id.download(f"{user_download_dir}/{file_id.file_id}.{file_extension}")
    except:
        file_download_path = None
        raise Exception(f"Couldn't download the file with file_id: {file_id}")

    return file_download_path
